src/answers-headless.ts

```diff
@@ -75,12 +76,21 @@ export default class AnswersHeadless {
   }
 
   /**
-   * Sets {@link VerticalSearchState.verticalKey} to the specified key.
+   * Sets up Headless to manage the vertical indicated by the verticalKey.
    *
    * @param verticalKey - The vertical key to set
    */
-  setVerticalKey(verticalKey: string): void {
+  setVertical(verticalKey: string): void {
     this.stateManager.dispatchEvent('vertical/setVerticalKey', verticalKey);
+    this.stateManager.dispatchEvent('meta/setSearchType', SearchTypeEnum.Vertical);
```

Comment:
You mentioned before that we can derive whether or not it's universal/vertical search based on verticalKey. so if setVertical and setUniversal properly update verticalKey, do we still need SearchTypeEnum?

Reply:
Yes, we could just use the verticalKey for this, however this function allows us to add additional functionality such as clearing the state when `setVertical` or `setUniversal` is called. Currently we handle that in the sample app, however product has mentioned it might make sense in Headless as well. That functionality isn't part of this PR, but we may add it in the future. Clearing parts of the state is outside the scope of `setVerticalKey`, but it would make sense in `setVertical`. I also think that checking the searchType is a little clearer than checking whether or not the verticalKey exists